src/hooks/useKanban.js

```diff
@@ -0,0 +1,301 @@
+import { useState, useEffect } from 'react';
+import { db } from '../firebase';
+import { 
+  collection, 
+  doc, 
+  setDoc, 
+  deleteDoc, 
+  onSnapshot,
+  query,
+  orderBy,
+  writeBatch,
+  getDoc
+} from 'firebase/firestore';
+import { useAuth } from '../contexts/AuthContext';
+
+export function useKanban() {
+  const [columns, setColumns] = useState({});
+  const [boardTitle, setBoardTitle] = useState('Kanban Board');
+  const [loading, setLoading] = useState(true);
+  const { user } = useAuth();
+
+  // Escuta mudanças nas colunas e no título
+  useEffect(() => {
+    const boardRef = doc(db, 'boards/main');
+    const columnsRef = collection(boardRef, 'columns');
+    const q = query(columnsRef, orderBy('order'));
+    
+    // Carregar título inicial
+    getDoc(boardRef).then((doc) => {
+      if (doc.exists() && doc.data().title) {
+        setBoardTitle(doc.data().title);
+      }
+    });
+
+    // Escutar mudanças no título
+    const unsubscribeBoard = onSnapshot(boardRef, (doc) => {
+      if (doc.exists() && doc.data().title) {
+        setBoardTitle(doc.data().title);
+      }
+    });
+
+    // Escutar mudanças nas colunas
+    const unsubscribeColumns = onSnapshot(q, (snapshot) => {
+      const newColumns = {};
+      snapshot.forEach((doc) => {
+        newColumns[doc.id] = { id: doc.id, ...doc.data() };
+      });
+      setColumns(newColumns);
+      setLoading(false);
+    }, (error) => {
+      console.error('Erro ao carregar dados:', error);
+      setLoading(false);
+    });
+
+    return () => {
+      unsubscribeBoard();
+      unsubscribeColumns();
+    };
+  }, []);
+
+  // Atualiza o título do board
+  const updateBoardTitle = async (newTitle) => {
+    if (!user) return;
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const updateBoardTitle = async (newTitle) => {
  if (!user) return;
  if (Date.now() - lastUpdate < RATE_LIMIT_MS) {
    throw new Error('Please wait before updating again');
  }
  lastUpdate = Date.now();
```
Consider implementing rate limiting for Firebase operations to prevent excessive writes and potential abuse.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;